api/v1beta1/cluster_types.go

```diff
@@ -328,11 +328,11 @@ type ClusterStatus struct {
 
 	// InfrastructureReady is the state of the infrastructure provider.
 	// +optional
-	InfrastructureReady bool `json:"infrastructureReady"`
+	InfrastructureReady bool `json:"infrastructureReady,omitempty"`
 
 	// ControlPlaneReady defines if the control plane is ready.
 	// +optional
-	ControlPlaneReady bool `json:"controlPlaneReady"`
+	ControlPlaneReady bool `json:"controlPlaneReady,omitempty"`
```

Comment:
Is this actually a wanted change? Seeing `ControlPlaneReady: false` in a kubectl get would probably give more signal to new users that don't know this field exist, or for existing users that are watching for changes and need to understand what's the state of things.

Reply:
Same applies for other, similar fields

Reply:
Thx for bringing this up. I'll bring the discussion back to the issue and close this PR for now